scrc/preprocessors/extractors/spider_specific/section_splitting_functions.py

```diff
@@ -25,13 +27,104 @@ def XX_SPIDER(decision: Union[bs4.BeautifulSoup, str], namespace: dict) -> Optio
     pass
 
 
+def VD_Omni(decision: Union[bs4.BeautifulSoup, str], namespace: dict) -> Optional[Dict[Section, List[str]]]:
+
+
+    def get_paragraphs(soup):
+        """
+        Get composition of the decision
+        :param soup:
+        :return:
+        """
+        # by some bruteforce I have found out that the html files in VD_Omni all have a single div with one of the
+        # following classes:
+        possible_main_div_classes = ["WordSection1", "Section1"]
+        # Iterate over this list and find out the current decision is using which class type
+        for main_div_class in possible_main_div_classes:
+
+            div = soup.find_all("div", class_=main_div_class)
+            if (len(div) != 0):
+                break
+        # We expect to have only one main div
+        assert len(div) == 1
+        # If the divs is empty raise an error
+        if len(div) == 0:
+            message = f"The main div has an unseen class type"
+            raise ValueError(message)
+        paragraphs = []
+        # paragraph = None
+        for element in div:
+            if isinstance(element, bs4.element.Tag):
+                text = str(element.string)
+                # This is a hack to also get tags which contain other tags such as links to BGEs
+                if text.strip() == 'None':
+                    text = element.get_text()
+                paragraph = clean_text(text)
+                if paragraph not in ['', ' ', None]:  # discard empty paragraphs
+                    paragraphs.append(paragraph)
+        return paragraphs
+    def get_composition_candidates(paragraphs, cm_RegEx, cm_end_RegEx):
+
+        # did we miss composition of some of the decisions?
+        missed_cnt = 0
+        composition_candidate = None
+        for paragraph in paragraphs:
+            cm = cm_RegEx.search(paragraph)
+            cm_end = cm_end_RegEx.search(paragraph)
+            # If we did not find the RegEx in the paragraph
+            if cm is None or cm_end is None:
+                missed_cnt += 1
+                continue
+            composition_candidate = paragraph[cm.start():cm_end.start()]
+
+            # Store all the compositions in a signle list
+
+        return composition_candidate, missed_cnt
```

Comment:
Why is this here? In the section splitting the goal is not to extract any court composition.

Reply:
I see your point. I have a question: Are you talking about the 'name' of this method? Or are you also concerned with the functionality?

If it is the first, I agree with you.

Otherwise, It would have been ideal if there was a section splitting function available for the VD_Omni, and I could just process the header. However, this is not the case, and therefore, I am trying to build a dummy section splitter that provides the necessary functionalities for court composition extraction. 

Would you please let me know what you think?